src/wallet/rpc/spend.cpp

```diff
@@ -1385,7 +1385,9 @@ RPCHelpMan sendall()
                     total_input_value += tx->tx->vout[input.prevout.n].nValue;
                 }
             } else {
-                for (const COutput& output : AvailableCoins(*pwallet, &coin_control, fee_rate, /*nMinimumAmount=*/0).All()) {
+                AvailableCoinsParams coins_params;
```

Comment:
nit: Regarding consistency for variable names, I think the `filter_coins` name that you use for the same variable in `wallet/rpc/coins.cpp` is better, or maybe a mixture of your current names: `coins_filter_params`.

Talking about naming, although `AvailableCoinsParams` is only used in the `AvailableCoins` function (hence the naming I guess), I think the name could do a better job describing the content. May I suggest `CoinFilter(ing)Params`, which also follows the naming scheme used in `CoinSelectionParams`.
https://github.com/bitcoin/bitcoin/blob/3f1f5f6f1ec49d0fb2acfddec4021b3582ba0553/src/wallet/coinselection.h#L116

Reply:
thought about it too. I tried to use a distinctive name for this struct because we do have some "coins filtering" args inside `CoinControl` as well.

But yeah, probably the best is to use the `CoinFilterParams` naming for this new struct and, in a future PR, instead of passing `CoinControl` to `AvailableCoins`, only pass `CoinFilterParams` (which will need to include what is inside `CoinControl`).

Reply:
I have to admit that I haven't look deep into the usage of CoinControl yet, but I I don't see such an issue on the distinction that `CoinFilterParams` are all the filters that are not part of `CoinControl`. Cause CoinControl is also a user-facing concept.

Reply:
What I meant is that, ideally, we should only provide `CoinFilterParams` to `AvailableCoins` and not two structs that contain filtering args. But that can be done later (I'm removing the preset inputs skip from `AvailableCoins` inside #25685).